### PR TITLE
Update localizations button

### PR DESCRIPTION
**Pull request by stalhandske**
Added a "Update Localizations" button to the Yarn Importer Editor. It goes through each string table CSVs in `localizations`, compares it to the base string table, merges them based on a set of rules, and creates updated versions of the localizations in seperate files.

Specifics on the merge rules used can be found as comments in the code.

### CODE REVIEW

File: Unity/Assets/YarnSpinner/Editor/YarnImporterEditor.cs [hunk 1]
```diff
@@ -160,4 +173,229 @@ public class YarnImporterEditor : ScriptedImporterEditor {
...
+            memoryStream.Position = 0;
+
+            using (var reader = new StreamReader(memoryStream))
+            {
+                return reader.ReadToEnd();
+            }
+        }
+    }
+    
+    public struct CsvEntry
```
Comment (desplesda) [inline Unity/Assets/YarnSpinner/Editor/YarnImporterEditor.cs:393]: If this struct isn't used outside this file, can it be private?

File: Unity/Assets/YarnSpinner/Editor/YarnImporterEditor.cs [hunk 2]
```diff
@@ -160,4 +173,229 @@ public class YarnImporterEditor : ScriptedImporterEditor {
...
+                if (!didFindInNew)
+                {
+                    oldIndex++;
+                }
+            }
+        }
+
+        mergedEntries.Sort((a, b) => a.lineNumber.CompareTo(b.lineNumber));
+
+        //foreach (CsvEntry e in mergedEntries)
```
Comment (desplesda) [inline Unity/Assets/YarnSpinner/Editor/YarnImporterEditor.cs:359]: Can you remove this commented-out debugging code?

File: Unity/Assets/YarnSpinner/Editor/YarnImporterEditor.cs [hunk 3]
```diff
@@ -160,4 +173,229 @@ public class YarnImporterEditor : ScriptedImporterEditor {
 
         AssetDatabase.ImportAsset(assetPath);
     }
+
+    void UpdateLocalizations(string newBaseCsv)
```
Comment (desplesda) [inline Unity/Assets/YarnSpinner/Editor/YarnImporterEditor.cs:177]: The documentation is really comprehensive in the second half of `MergeStringTables`, but there's a bit less in the first half and in `UpdateLocalizations`. Can you expand on the comments a bit?

Comment (stalhandske) [reply Unity/Assets/YarnSpinner/Editor/YarnImporterEditor.cs:178]: Yep, new commit up with more comments and fixes

**Pull request merged**